servers/src/main/java/tachyon/worker/block/evictor/LRUEvictor.java

```diff
@@ -84,7 +86,32 @@ public EvictionPlan freeSpace(long availableBytes, BlockStoreLocation location)
 
     // enough free space
     if (dirCandidates.maxAvailableBytes() >= toEvictBytes) {
+      // candidate blockIds for eviction from current tier
       toEvict = dirCandidates.toEvict();
+      // move as many blocks to next tier as possible
+      int nextTierLevel = mMeta.getBlockMeta(toEvict.get(0)).getBlockLocation().tierLevel() + 1;
+      List<StorageTier> tiers = mMeta.getTiers();
+      int nTotalTiers = tiers.size();
+      while (nextTierLevel < nTotalTiers && toEvict.size() > 0) {
+        // find space in next tier
+        List<StorageDir> dirs = tiers.get(nextTierLevel).getStorageDirs();
+        for (StorageDir dir : dirs) {
+          BlockStoreLocation dest = dir.toBlockStoreLocation();
+          Iterator<Long> blocks = toEvict.iterator();
+          long remainBytes = dir.getAvailableBytes();
+          while (blocks.hasNext() && remainBytes >= 0) {
+            long blockId = blocks.next();
+            long blockSize = mMeta.getBlockMeta(blockId).getBlockSize();
+            if (blockSize <= remainBytes) {
+              // the block can be moved to the dir
+              toMove.add(new Pair<Long, BlockStoreLocation>(blockId, dest));
+              blocks.remove();
+              remainBytes -= blockSize;
+            }
+          }
+        }
+        nextTierLevel ++;
+      }
```

Comment:
Could you add a check to validate all the blocks are in the tiered store before returning the plan?